linodecli/plugins/obj.py

```diff
@@ -190,9 +204,48 @@ def upload_object(client, args):
         policy = 'public-read' if parsed.acl_public else None
         k.set_contents_from_filename(file_path, cb=_progress, num_cb=100, policy=policy)
 
+    for filename, file_path, file_size in to_multipart_upload:
+        _do_multipart_upload(bucket, filename, file_path, file_size)
+
     print('Done.')
 
 
+def _do_multipart_upload(bucket, filename, file_path, file_size):
+    """
+    Handles the internals of a multipart upload for a large file.
+
+    :param bucket: The bucket to upload the large file to
+    :type bucket: Boto bucket
+    :param filename: The name of the file to upload
+    :type filename: str
+    :param: file_path: That absolute path to the file we're uploading
+    :type file_path: str
+    :param file_size: The size of this file in bytes (used for chunking)
+    :type file_size: int
+    """
+    upload = bucket.initiate_multipart_upload(filename)
+
+    num_chunks = int(math.ceil(file_size / MULTIPART_UPLOAD_CHUNK_SIZE))
+    upload_exception = None
+
+    print("{} ({} parts)".format(filename, num_chunks))
+
+    try:
+        with open(file_path, "rb") as f:
+            for i in range(num_chunks):
+                print(" Part {}".format(i+1))
+                upload.upload_part_from_file(f, i+1, cb=_progress, num_cb=100, size=MULTIPART_UPLOAD_CHUNK_SIZE)
+    except Exception as e:
+        upload_exception = e
+    finally:
```

Comment:
why does this live in the finally block and not in `except`?

Reply:
Originally there were other ways it could happen, but I guess they got removed.  I'll clean it up